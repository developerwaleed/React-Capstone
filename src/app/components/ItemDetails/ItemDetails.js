import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchItemDetail, removeSelectedItems } from '../../../redux/products/products';
import styles from './ItemDetails.module.css';

const ItemDetails = () => {
  const { productId } = useParams();
  const product = useSelector((state) => state.product);
  const {
    image, title, price, category, description,
  } = product;
  const dispatch = useDispatch();
  useEffect(() => {
    if (productId && productId !== '') dispatch(fetchItemDetail(productId));
    return () => {
      dispatch(removeSelectedItems());
    };
  }, []);
  return (
    <div className={styles.productContainer}>
      {Object.keys(product).length === 0 ? (
        <div className={styles.loadingContainer}><div className="customLoader" /></div>
      ) : (
        <div className={styles.productDetailContainer}>
          <div className={styles.detailsImg}>
            <img className={styles.itemImage} src={image} alt={title} />
          </div>
          <div className={styles.detailsContent}>
            <h2 className={styles.itemName}>{title}</h2>
            <h3>
              <p className={styles.itemPrice}>
                $
                {price}
              </p>
            </h3>
            <h4 className={styles.itemCategory}>{category}</h4>
            <p className={styles.itemDescription}>{description}</p>
          </div>
        </div>
      )}
    </div>
  );
};

export default ItemDetails;
