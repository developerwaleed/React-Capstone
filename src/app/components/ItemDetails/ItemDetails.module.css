.loadingContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 10rem 0;
}

.productDetailContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 4% 6%;
  background-color: #fff;
  border-radius: 20px;
  min-height: 600px;
  box-shadow: 6px 8px 9px -2px rgba(0, 0, 0, 0.34);
}

.detailsImg img {
  display: block;
  margin: 0 auto;
  width: 65%;
}

.itemImage {
  display: block;
  margin: 0 auto;
  width: 35%;
}

.detailsContent {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.itemName {
  font-size: 1.3rem;
  text-align: center;
}

.itemPrice {
  font-size: 1.4rem;
}

.itemCategory {
  font-size: 0.6rem;
  align-self: center;
  background-color: #4369b2;
  background-color: var(--color-2);
  border-radius: 20px;
  color: #fff;
  font-size: 0.5rem;
  font-weight: 500;
  padding: 0.3rem 0.7rem;
  text-transform: uppercase;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.itemDescription {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 0.7rem 0.5rem;
  row-gap: 0.5rem;
  text-align: center;
  color: #000;
  font-size: 0.8rem;
  opacity: 0.8;
  text-transform: none;
}

@media screen and (min-width: 768px) {
  .productDetailContainer {
    min-height: 600px;
    max-width: 700px;
    margin: 3% auto;
  }

  .itemImage {
    width: 45%;
  }

  .itemName {
    font-size: 1.1rem;
  }

  .itemPrice {
    font-size: 1.7rem;
  }

  .itemCategory {
    font-size: 0.7rem;
  }

  .itemDescription {
    font-size: 1rem;
  }
}
