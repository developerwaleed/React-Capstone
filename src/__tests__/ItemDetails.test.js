import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ItemDetails from '../app/components/ItemDetails/ItemDetails';
import store from '../redux/configureStore';

it('renders properly', () => {
  const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <ItemDetails />
      </Provider>
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
