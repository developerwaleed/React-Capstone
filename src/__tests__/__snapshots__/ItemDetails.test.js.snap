// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="productContainer"
>
  <div
    className="loadingContainer"
  >
    <div
      className="customLoader"
    />
  </div>
</div>
`;
